app/main/viewmanager.ts

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+import { BrowserWindow, ipcMain } from 'electron';
+import { View, ViewProps } from './view';
+
+import ConfigUtil = require('../renderer/js/utils/config-util');
+
+class ViewManager {
+	views: { [key: number]: View };
+	selectedIndex: number;
+
+	constructor() {
+		this.views = {};
+		this.selectedIndex = 0;
+		this.registerIpcs();
+	}
+
+	registerIpcs(): void {
+		ipcMain.on('view-create', (e: Event, props: ViewProps) => {
```

Comment:
This sounds, and reads for that matter, better as `create-view`, `select-view` and `destroy-view`?